src/facade/dragonfly_connection.cc

```diff
@@ -1429,24 +1416,16 @@ auto Connection::IoLoop() -> variant<error_code, ParserStatus> {
                               [&]() { io_buf_.Reserve(std::min(max_iobfuf_len, parser_hint)); });
         }
 
-        // If we got a partial request and we couldn't parse the length, just
-        // double the capacity.
         // If we got a partial request because iobuf was full, grow it up to
         // a reasonable limit to save on Recv() calls.
-        if (io_buf_.AppendLen() < 64u || (is_iobuf_full && capacity < 4096)) {
+        if (is_iobuf_full && capacity < max_iobfuf_len / 2) {
```

Comment:
If `is_iobuf_full` && `capacity > max_iobuf_len / 2` it means we can't grow. But what if the buffer is full and the command is partial ? Won't we need the condition below you removed ? (because the request is indeed to large). What am I missing here ?

Reply:
I've changed the implementation of RedisParser - it now consumes all the data - even for partial request. Therefore, it will consume the data in the buffer and the connection will continue reading the rest in max_iobfuf_len chunks